### PR TITLE
Expose workflow history size and count to client

### PR DESCRIPTION
**Pull request by timl3136**
**What changed?**
Added fields regarding workflow history size and count in PollForDecisionResponse to be exposed to cadence-client.
History size will be exposed in unit of bytes and history count will be the number of completed events.
Added fields sending history data from cadence-history to cadence-matching when matching service calls the persistence layer.
Updated dependency for cadence-idl to include new fields in idl in sync with cadence-client.

**Why?**
There have been requests from customers wanting history information regarding workflow to be exposed in client so that they can optimize their workflow through methods like continueAsNew when approaching the history size limit.
Resolved cadence-client issue [#917 ](https://github.com/uber-go/cadence-client/issues/917)

**How did you test it?**
Standard unit test as well as tested in development environment.

**Potential risks**
Since we have added new fields to thrift and proto layer, there should not be any breaking change since new fields will be defaulted to 0. However, there remains the possibility that the change will result in thrift/proto field mismatch between serve and client hence result in information loss.

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: client/history/client.go [hunk 1]
```diff
@@ -127,6 +128,7 @@ func (c *clientImpl) GetMutableState(
 	if err != nil {
 		return nil, err
 	}
+	c.logger.Info(fmt.Sprintf("History ClientImpl History size %d", response.HistorySize))
```
Comment (Shaddoll) [inline client/history/client.go:131]: can you remove this log?

File: common/types/matching.go [hunk 2]
```diff
@@ -455,6 +456,14 @@ func (v *MatchingPollForDecisionTaskResponse) GetBranchToken() (o []byte) {
 	return
 }
 
+// GetHistorySize is an internal getter of returning the history size in bytes
+func (v *MatchingPollForDecisionTaskResponse) GetHistorySize() (o int64) {
```
Comment (Shaddoll) [inline common/types/matching.go:460]: The mismatch between field name and getter name is confusing.

File: service/matching/matchingEngine.go [hunk 3]
```diff
@@ -488,12 +495,14 @@ pollLoop:
 				StickyExecutionEnabled:    isStickyEnabled,
 				WorkflowExecutionTaskList: mutableStateResp.TaskList,
 				BranchToken:               mutableStateResp.CurrentBranchToken,
+				HistorySize:               mutableStateResp.HistorySize,
```
Comment (Shaddoll) [inline service/matching/matchingEngine.go:498]: are you exposing history size only for query tasks? what if users don't do queries?

File: common/types/history.go
Comment (Shaddoll) [inline common/types/history.go:1]: Could you update the unit tests of this package to make sure the new field is included by mapper?

File: common/util.go [hunk 4]
```diff
@@ -413,6 +413,7 @@ func CreateMatchingPollForDecisionTaskResponse(historyResponse *types.RecordDeci
 		ScheduledTimestamp:        historyResponse.ScheduledTimestamp,
 		StartedTimestamp:          historyResponse.StartedTimestamp,
 		Queries:                   historyResponse.Queries,
+		TotalHistoryBytes:         historyResponse.HistorySize,
```
Comment (Shaddoll) [inline common/util.go:416]: Do we expose historyCount?

File: service/matching/matchingEngine.go (hunk 3 above)
Comment (timl3136) [reply service/matching/matchingEngine.go:498]: For queue task, history size is exposed through the existing GetMutableState function calling (here). For non-query task, history size is exposed through the RecordDecisionTaskStarted function.

File: client/history/client.go (hunk 1 above)
Comment (timl3136) [reply client/history/client.go:131]: Resolved

File: common/util.go (hunk 4 above)
Comment (timl3136) [reply common/util.go:416]: Yes, the historyCount is equivalent to NextEventID - 1

File: common/types/matching.go (hunk 2 above)
Comment (timl3136) [reply common/types/matching.go]: Resolved

File: common/types/history.go
Comment (timl3136) [reply common/types/history.go:1]: Resolved

**Review by Shaddoll — APPROVED** [review]

**Pull request merged**